### ISSUE TITLE
IE8 can drag but not drop

### ISSUE THREAD

**Issue by eskarinna** [bevacqua/dragula#177]
I have the latest version of dragula, es5 and everything else. when I am in IE8 I can start dragging the item however it doesn't drop anywhere on the page. After upgrading to the latest version I also started getting a tyoe mismatch error pretty much on everything I click and only on the page that my drag and drop widget is located. 
<img width="731" alt="dragulaerror" src="https://cloud.githubusercontent.com/assets/12157544/9430168/a91616da-49b0-11e5-9786-9cb171992f20.png">

Also if I drag the image anywhere else but the destination content including back to where it was before the drag the html seems to get all whacky. SS attached

<img width="750" alt="2015-08-23_16-00-15" src="https://cloud.githubusercontent.com/assets/12157544/9430170/b1a15288-49b0-11e5-8ad1-7c2b7eb6d832.png">

My js code is:

```js
// --------------------------------------------------
// Drag+drop things around
// --------------------------------------------------
var gallery_content = document.querySelector('.widget-gallery-content');
var gallery_content_list = document.querySelector('.widget-gallery-content .draggable-group');
var gallery_trash = document.querySelector('.widget-gallery-trash');
var source_list = document.querySelector('.body_content .draggable-group');

$(document).ready(function () {

    check_bookmarkable();

    var dashboardApps = [];

    $('.body_content .draggable-group').find('[data-resourceid]').each(function () {
        dashboardApps.push($(this).data('resourceid'));
    })

    var asideApps = [];
    intersection = [];

    $('.widget-gallery-content .draggable-group').find('[data-resourceid]').each(function () {
        asideApps.push($(this).data('resourceid'));
    })

    refreshDashboard();

    function refreshDashboard() {

        for (i in dashboardApps) {
            for (j in asideApps) {
                if (dashboardApps[i] == asideApps[j]) {

                    intersection.push(dashboardApps[i]);

                    $('.widget-gallery-content .draggable-group').find('[data-resourceid="' + dashboardApps[i] + '"]').hide();

                }
            }
        }

    }

   
    if (typeof (dragula) !== 'undefined' && (gallery_content && gallery_content_list && gallery_trash && source_list)) {
      
        var widget_drag = dragula(
			[
                gallery_content_list,
				source_list,
				gallery_trash
			],
			{
			    moves: function (el, container, handle) {
			        if (drag_enabled) {
			            if (
							el.className.match(/(?:^|\s)widget-item(?!\S)/) ||
							el.className.match(/(?:^|\s)card(?!\S)/)
						) {
			                return true;
			            }
			        }

			        return false;
			    },
			    revertOnSpill: true
			}
		);

        // show the trash space when starting drag from body_content
        widget_drag.on('drag', function (el, container) {
            // if starting from body_content
            if (container == source_list) {
                gallery_content.className = gallery_content.className.replace(/(?:^|\s)active(?!\S)/g, '');
                gallery_trash.className += ' active';
            } else {
                $(el).css({ position: 'relative' });
            }
        });

        // hide the trash after drag ends
        widget_drag.on('dragend', function (el) {
            gallery_trash.className = gallery_trash.className.replace(/(?:^|\s)active(?!\S)/g, '');
            gallery_content.className += ' active';

            el.className = el.className.replace(/(?:^|\s)skeleton(?!\S)/g, '');

            /**
             * Check widget panel for any post processing
             */
            // Add filters back if widget is drug back to panel
            var widgets = $('.aside-tiles-container .widget-item');
            if (widgets.length) {
                widgets.each(function (key, widget) {
                    var filterValue = $(widget).data('parent');
                    var option = $('option[value="' + filterValue + '"]').show();
                });
            } else {
                $('#widget-filter option').hide();
                $('#widget-filter option[value="*"]').show();
                $('#widget-filter').val('*');
            }

            // Remove filter if no widgets left in panel
            $('#widget-filter option').each(function (key, option) {
                if ($(option).val() !== '*') {
                    if (!$('.widget-item[data-parent="' + $(option).val() + '"]').length) {
                        $(option).hide();

                        if ($(option).is(':selected')) {
                            $('#widget-filter').val('*');
                        }
                    }
                }
            });
            $("#widget-filter").trigger("chosen:updated");
            applyFilter();
        });

        // shadow changes based on location
        widget_drag.on('shadow', function (el, container) {
            // if the shadow is in the body_content
            if (container == source_list) {
                // if it's a widget add the proper classes to make it a card
                if (el.className.match(/(?:^|\s)widget-item(?!\S)/)) {
                    el.className = "card application-card";
                }

                // if item doesn't already have the skeleton class add it
                if (!el.className.match(/(?:^|\s)skeleton(?!\S)/)) {
                    el.className += " skeleton";
                }

            }

            if (container == gallery_content_list || container == gallery_trash) {
                // if it's a card add the proper classes to make it a widget
                if (el.className.match(/(?:^|\s)card(?!\S)/)) {
                    el.className = "widget-item-tile";
                }
            }

            // no shadow if we're over the trash
            // :: highlight trash when over, remove when not
            if (container == gallery_trash) {
                el.style.display = "none";
                container.className += " highlight";

            } else {
                el.style.display = "block";
                gallery_trash.className = gallery_trash.className.replace(/(?:^|\s)highlight(?!\S)/g, '');
            }

        });

        widget_drag.on('drop', function (el, container, source) {
            // if dropping on trash don't keep it and move it to the proper location
            if (container == gallery_trash) {
                var item = $(el).detach();
                item.addClass('widget-item aside-tile');
                item.show();

                item.appendTo(gallery_content_list);


                //this is the delete function

                var resourceId = $(el).data('resourceid');

                $.ajax({
                    type: 'POST',
                    url: '/data/edit-dashboard',
                    data: {
                        'applicationToDelete': resourceId
                    },

                    success: function (data) {
                        var json = JSON.parse(data);
                    }
                });

            } else {

                //saving the element in the database

                var resourceId = $(el).data('resourceid');

                $.ajax({
                    type: 'POST',
                    url: '/data/edit-dashboard',
                    data: {
                        'applicationToAdd': resourceId
                    },

                    success: function (data) {
                        var json = JSON.parse(data);
                    }
                });
            }

            applyFilter();
        });
    }
});
```

**Comment by Balkoth** [comment]
I have the same issue since commit b0e5c4fc1017e1e33e2013ec8e47d43a46164c86.
As a workaround, you can update the function `removeEventHard` by adding a test like this:
```js
function removeEventHard (el, type, fn) {
    var ev = unwrap(el, type, fn);
    if (ev) return el.detachEvent('on' + type, ev);
}
```
But I'm not sure it's the best solution.

**Comment by eskarinna** [comment]
Thank you if that works, know you saved a life! I cried lol

**Comment by bevacqua** [comment]
Fixed in `3.0.4`

**Comment by eskarinna** [comment]
It worked like a charm balloh!

Ty for the fix bevacqua
